libmamba/src/core/util.cpp

```diff
@@ -1487,4 +1488,33 @@ namespace mamba
         return std::string((const char*) output.data());
     }
 
+    std::optional<std::string> proxy_match(const std::string& url)
+    {
+        auto& proxies = Context::instance().proxy_servers;
+        if (proxies.empty())
```

Comment:
Nit: Should be ok to drop this case.

Reply:
I would prefer to leave that in for readability. It's not immediately obvious that empty `proxies` leads to match staying the default constructor value, which is empty string, which in turn fails the `.size !=0` check.